Custom module improvements

dicarlosystems opened the pull request:
Just opening this with some improvements for custom module integration.  I will be pushing additional changes to this branch as ideas are fleshed out more.

hillelcoren commented:
These changes look great!

dicarlosystems commented:
Thanks!  My only concern is with the setup stuff - it seems clunky and inelegant.  But I was thinking it's better to have something working and tested a bit more before trying to make it super elegant, especially if it doesn't satisfy the need.

dicarlosystems commented:
Also, I was thinking about the view components PR - I wonder if those changes should just be rolled up under this broader PR?

hillelcoren commented:
It's fine to leave it as it, maybe just keep it in mind for the future.

Smaller PR's are easier to review/merge.

dicarlosystems commented:
Sounds good.

hillelcoren commented:
Let me know if I should merge this...

dicarlosystems commented:
I think it's good to merge.  I will be doing some more testing of this functionality as part of some modules I'm working on, so it's a good test bed.

dicarlosystems commented:
Shoot, meant to add that.  Thanks for catching it!  I’ll push the change.

> On Sep 9, 2018, at 01:19, Hillel Coren <notifications@github.com> wrote:
> 
> Can you please make one change before we merge, can you add isSelfHost checks in the list.blade file for the module related code.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

dicarlosystems commented:
I tweaked the settings implementation to simplify the setup and updated the docs.  Think it's good now.

The main changes are convention and UI cleanup:
- if there are no settings defined for at least one module, the section is not displayed;
- the module settings are defined as a settings.blade.php file in the module;
- I removed the explicit definition of the routes; the developer can choose to register and implement the routes however they like;
- if a module defines that it has settings but does not include a settings blade template, the entry is not included.

Wouter0100 commented on resources/views/list.blade.php:
```diff
@@ -46,6 +54,10 @@
 
+	@if (Utils::isSelfHost())
+    	@stack('top_right_buttons')
+    @endif
```
Indentation error? :)

dicarlosystems replied:
Weird, my local copy doesn't have the indentation problem.

dicarlosystems replied:
Looks like it was an issue with tabs versus spaces.

turbo124 commented:
@dicarlosystems 

There seems to be a couple of issues with the PR 

It looks like some controllers are missing the

          'showModuleSettings' => Utils::hasModuleSettings(),

I can see taxratecontroller is failing. are you able to have a look at this?

dicarlosystems commented:
I will take a look.

dicarlosystems commented:
@turbo124 I forgot to add the settings to some of the sub-settings views, e.g. tax rates, users, etc.  I'm making the changes and will issue a new PR to fix the regressions.
